rmgpy/molecule/filtration.py

```diff
@@ -210,40 +211,37 @@ def charge_filtration(filtered_list, charge_span_list):
-            # Find unique radical sites in charged_list and append to unique_charged_list:
-            unique_charged_list = []
-            for mol in charged_list:
-                for atom in mol.vertices:
-                    if atom.radicalElectrons and int(atom.sortingLabel) not in rad_sorting_list:
-                        unique_charged_list.append(mol)
-
-            # Charge stabilization considerations for the case where there are several charge span layers and the
-            # species is a radical are checked here for filtered_list and unique_charged_list separately.
-            filtered_list = stabilize_charges_by_electronegativity(filtered_list)
-            filtered_list = stabilize_charges_by_proximity(filtered_list)
-            if unique_charged_list:
-                unique_charged_list = stabilize_charges_by_electronegativity(unique_charged_list, allow_empty_list=True)
-                unique_charged_list = stabilize_charges_by_proximity(unique_charged_list)
-                filtered_list.extend(unique_charged_list)
-        else:
-            # There are structures with different charge spans, but the species is not a radical.
-            # Additional charge span levels are removed.
-            filtered_list = [filtered for index, filtered in enumerate(filtered_list) if
-                             charge_span_list[index] == min_charge_span]
+        charged_list = [filtered_mol for index, filtered_mol in enumerate(filtered_list) if
+                        charge_span_list[index] == min_charge_span + 1]  # save the 2nd charge span layer
+        filtered_list = [filtered_mol for index, filtered_mol in enumerate(filtered_list) if
+                         charge_span_list[index] == min_charge_span]  # the minimal charge span layer
+        # Find the radical and multiple bond sites in all filtered_list structures:
+        rad_sorting_list = []  # sortingLabels for radical sites
+        mul_bond_sorting_list = []  # sortingLabels for multiple bind sites in the form of (atom1,atom2) tuples
+        for mol in filtered_list:
+            for atom in mol.vertices:
+                if atom.radicalElectrons and int(atom.sortingLabel) not in rad_sorting_list:
+                    rad_sorting_list.append(int(atom.sortingLabel))
+                for atom2, bond in atom.bonds.iteritems():
+                    # check if bond is multiple, store only from one side (atom1 < atom2) for consistency
+                    if atom2.sortingLabel > atom.sortingLabel and bond.isDouble() or bond.isTriple():
+                        mul_bond_sorting_list.append((int(atom.sortingLabel), int(atom2.sortingLabel)))
+        # Find unique radical and multiple bond sites in charged_list and append to unique_charged_list:
+        unique_charged_list = []
+        for mol in charged_list:
+            unique_charged_list.extend(find_unique_sites_in_charged_list(mol, rad_sorting_list, mul_bond_sorting_list))
```

Comment:
The approach here seems a bit complex. Could you move the outer for loop into `find_unique_sites_in_charged_list` and have it directly return `unique_charged_list`? Then I think you would be able to use append instead of extend.

Reply:
As discussed off-line, the code here is left as is since the helper function's `return` assists in breaking out of two `for` loops